src/NHibernate.Test/Linq/LinqQuerySamples.cs

```diff
@@ -1558,14 +1672,27 @@ public void DLinqJoin8()
 				from o in ords
 				select new {c.ContactName, o.OrderId, z};
 
-			ObjectDumper.Write(q);
+			using (var sqlSpy = new SqlLogSpy())
+			{
+				ObjectDumper.Write(q);
+
+				var sql = sqlSpy.GetWholeLog();
+				Assert.That(GetTotalOccurrences(sql, "inner join"), Is.EqualTo(1));
+			}
 		}
 
 		[Category("JOIN")]
-		[Test(Description = "This sample shows a group join with a composite key.")]
-		public void DLinqJoin9()
+		[TestCase(true, Description = "This sample shows a group join with a composite key.")]
+		[TestCase(false, Description = "This sample shows a group join with a composite key.")]
+		public void DLinqJoin9(bool useCrossJoin)
 		{
-			var expected =
+			if (useCrossJoin && !Dialect.SupportsCrossJoin)
+			{
+				Assert.Ignore("Dialect does not support cross join.");
+			}
+
+			ICollection expected, actual;
+			expected =
```

Comment:
If I'm not mistaken - this query loads a big bunch or records in multiple db roundtrips just to check total count of expected records. Not good. How about commenting this part (just in case test data is changed) and leave in test only check with hardcoded expected count (as you already did ) and  leave comment that this value can be obtained from this query.

Reply:
I agree, the query is now commented.